api/v1alpha2/azuremachine_conversion.go

```diff
@@ -72,6 +72,14 @@ func Convert_v1alpha2_AzureMachineSpec_To_v1alpha3_AzureMachineSpec(in *AzureMac
 		return err
 	}
 
+	if in.AvailabilityZone.ID != nil {
+		// If there is a AZ specified then copy it to failure domain
+		// and then nil the AZ
+		out.FailureDomain = in.AvailabilityZone.ID
+		out.AvailabilityZone.ID = nil
+		out.AvailabilityZone.Enabled = nil
```

Comment:
@detiber @CecileRobertMichon  - On the conversion v1alpha2->v1alpha3 does this make sense? If there is an AZ specified when we convert this gets moved to FailureDomain and the value in AvailabilityZone is removed. Of should they both be filled in?

Reply:
In order to facilitate the setting of Spec.FailureDomain when Spec.AvailabilityZone is set to get the data bubbled up to the Machine resource, this should probably be done in the AzureMachine controller rather than trying to handle it in the conversions. Especially since the field is being added after the initial v1alpha3 release and there will be v1alpha3 resources that will need to have the migration done.

Reply:
Thanks for your feedback on this, what you suggested made it simpler.